File: LV4/4.4/cd/cd.rb
loop do
  a, b = gets.split.map(&:to_i)
  break if a.zero? && b.zero?

  aa = (1..a).map { gets.to_i }
  idx = 0
  cnt = 0

  b.times do
    bb = gets.to_i
    idx += 1 while idx < a && bb > aa[idx]
    cnt += 1 if idx < a && bb == aa[idx]
  end

  puts cnt
end

File: LV4/4.4/cd/cd.py
# +----------+
# | TOO SLOW |
# +----------+
while True:
    a, b = [int(n) for n in input().split(' ')]
    if a == 0 and b == 0:
        break

    A = [int(input()) for _ in range(a)]
    idx = 0
    cnt = 0

    for _ in range(b):
        B = int(input())

        while idx < a and B > A[idx]:
            idx += 1

        if idx < a and B == A[idx]:
            cnt += 1

    print(cnt)
